frontend/test/pytest/test_transform.py

```diff
@@ -121,44 +166,896 @@ def qfunc(data, x, weights):
+            qml.RX(x[1], wires=1)
+            return [
+                qml.expval(qml.X(0) @ qml.Z(1) + 0.5 * qml.Y(1) + qml.Z(0)),
+                qml.expval(qml.X(1) + qml.Y(1)),
+            ]
+
+        return qfunc
+
+    qnode_control = qnode_builder("default.qubit")
+    qnode_backend = qnode_builder(backend)
+
+    jax_jit = jax.jit(qnode_control)
+    compiled = qjit(qnode_backend)
+
+    x = [np.pi / 4, np.pi / 4]
+    expected = jax_jit(x)
+    observed = compiled(x)
+    assert np.allclose(expected, observed)
+
+    _, expected_shape = jax.tree_util.tree_flatten(expected)
+    _, observed_shape = jax.tree_util.tree_flatten(observed)
+    assert expected_shape == observed_shape
+
+
+@pytest.mark.xfail(reason="Something in pyzx not JAX JIT compatible.")
+def test_to_zx(backend):
+    """Test to_zx"""
+
+    def qnode_builder(device_name):
+        """Builder"""
+
+        @qml.transforms.to_zx
+        @qml.qnode(qml.device(device_name, wires=2), interface="jax")
+        def qfunc(p):
+            qml.RZ(p[0], wires=1)
+            qml.RZ(p[1], wires=1)
+            qml.RX(p[2], wires=0)
+            qml.Z(0)
+            qml.RZ(p[3], wires=1)
+            qml.X(1)
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[1, 0])
+            return qml.expval(qml.Z(0) @ qml.Z(1))
+
+        return qfunc
+
+    qnode_control = qnode_builder("default.qubit")
+    qnode_backend = qnode_builder(backend)
+
+    jax_jit = jax.jit(qnode_control)
+    compiled = qjit(qnode_backend)
+
+    params = [5 / 4 * np.pi, 3 / 4 * np.pi, 0.1, 0.3]
+    expected = jax_jit(params)
+    observed = compiled(params)
+    assert np.allclose(expected, observed)
+
+    _, expected_shape = jax.tree_util.tree_flatten(expected)
+    _, observed_shape = jax.tree_util.tree_flatten(observed)
+    assert expected_shape == observed_shape
+
+
+def test_transpile(backend):
+    """Test transpile"""
+
+    def qnode_builder(device_name):
+        """Builder"""
+
+        @partial(qml.transforms.transpile, coupling_map=[(0, 1), (1, 3), (3, 2), (2, 0)])
+        @qml.qnode(qml.device(device_name, wires=4), interface="jax")
+        def qfunc():
+            qml.CNOT(wires=[0, 1])
+            qml.CNOT(wires=[2, 3])
+            qml.CNOT(wires=[1, 3])
+            qml.CNOT(wires=[1, 2])
+            qml.CNOT(wires=[2, 3])
+            qml.CNOT(wires=[0, 3])
+            return qml.probs(wires=[0, 1, 2, 3])
+
+        return qfunc
+
+    qnode_control = qnode_builder("default.qubit")
+    qnode_backend = qnode_builder(backend)
+
+    jax_jit = jax.jit(qnode_control)
+    compiled = qjit(qnode_backend)
+
+    expected = jax_jit()
+    observed = compiled()
+    assert np.allclose(expected, observed)
+
+    _, expected_shape = jax.tree_util.tree_flatten(expected)
+    _, observed_shape = jax.tree_util.tree_flatten(observed)
+    assert expected_shape == observed_shape
+
+
+@pytest.mark.xfail(reason="JIT and QJIT return different values.")
```

Comment:
How about qjit and plain pennylane?

Reply:
QJIT differs from plain PennyLane too. I will open a bug report for this.